napari_aicssegmentation/core/layer_reader.py

```diff
@@ -14,17 +17,26 @@ class LayerReader:
     def get_channels(self, layer: Layer) -> List[Channel]:
         """
         Get the list of image channels from a layer
-        TODO this is a workaround for now and we just guess the Channel dimension based on its
-             location for most ome tiffs
-        TODO use aicsimageio to read image from the source file path and get channel names
-             once Napari exposes Image layer source (next release)
 
         inputs:
             layer (Layer): the Napari layer to read data from
         """
         if layer is None:
             return None
 
+        if layer.source is not None and layer.source.path is not None:
+            try:
+                return self._get_channels_from_path(layer.source.path)
+            except Exception as ex:
+                log.warning(
+                    "Could not read image layer from source path even though a source path was provided."
+                    "Defaulting to reading from layer data (this is less accurate). \n"
+                    f"Error message: {ex}"
+                )
+
+        return self._get_channels_default(layer)
+
+    def _get_channels_default(self, layer: Layer) -> List[Channel]:
         img = AICSImage(layer.data)  # gives us a 6D image
 
         # we're expecting either STCZYX or STZCYX but we don't know for sure
```

Comment:
Okay I'm probably missing something but do we still not know this for sure?

Reply:
If we don't have the source image path to open (which should be the case if the layer wasn't loaded directly from a file), we're still stuck guessing.